### ISSUE TITLE
Can't create mosaic when geometry have different number of points ? 

### ISSUE THREAD

**Issue by vincentsarago** [developmentseed/cogeo-mosaic#81]
Bug found while working on #80 

```python
import requests
from pygeos import polygons
headers = {
    "Content-Type": "application/json",
    "Accept-Encoding": "gzip",
    "Accept": "application/geo+json",
}
data = requests.post("https://earth-search.aws.element84.com/v0/search", headers=headers).json()
dataset_geoms = polygons(
    [feat["geometry"]["coordinates"][0] for feat in data["features"]]
)
```

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-083d9d799f2d> in <module>
      1 dataset_geoms = polygons(
----> 2     [feat["geometry"]["coordinates"][0] for feat in data["features"]]
      3 )

pygeos/creation.py in polygons(shells, holes)
     88 
     89     if holes is None:
---> 90         return lib.polygons_without_holes(shells)
     91 
     92     holes = np.asarray(holes)

TypeError: One of the arguments is of incorrect type. Please provide only Geometry objects.
```

seems to be related to https://github.com/pygeos/pygeos/issues/138 

cc @kylebarron

**Comment by vincentsarago** [comment]
Quick fix ? 

```python
try:
    dataset_geoms = polygons(
        [feat["geometry"]["coordinates"][0] for feat in features]
    )
except TypeError:
    dataset_geoms = [
        [polygons(feat["geometry"]["coordinates"][0]) for feat in features]
    ]
```

**Comment by vincentsarago** [comment]
fixed in #82
